### PR TITLE
[SCHAINS-281] Report month by week

### PR DESCRIPTION
**Pull request by carlosper88**
This PR includes the changes requested in this ticket
https://horizenlabs.atlassian.net/browse/SCHAINS-281

It has also been added some testing and refactored some test directories

### CODE REVIEW

File: src/utils/Utils.ts [hunk 1]
```diff
@@ -13,12 +13,44 @@ export function getFormattedDate(date?: Date, format?: DateFormat): string {
     return ""
 }
 
+export function getWeekOfMonth(date) {
```
Comment (otoumas) [inline src/utils/Utils.ts:16]: Maybe we could rename this file DateUtils as all the methods are date related?

File: test/integration/initial_sync/report.tests.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            });
+        });
+        block.sidechainTransactions.forEach(tx => {
+            tx.newBoxes.forEach(box => {
+                nodeBoxes.push(box);
+            });
+        });
+    });
+    return nodeBoxes;
+};
```
Comment (otoumas) [inline test/integration/initial_sync/report.tests.ts:39]: Is this code the same as in boxes test class? Maybe we can move it to another helper/utils class and reduce duplication

File: test/integration/initial_sync/report.tests.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+    describe("'/report-month'", () => {
+        it("should return no from date sent error", async () => {
+            await axios.get(`${ params.explorerUrl }/report-month`)
+                .then(response => {
+                    expect(response.data).to.not.exist
+                })
+                .catch((error) => {
+                    expect(error.response.data).to.equal("Missing \"from\" date")
+                })
```
Comment (otoumas) [inline test/integration/initial_sync/report.tests.ts:51]: This is a little disconcerting is the first expect really happening? I mean at runtime does it really go through the then clause?
There is another way of testing the errors, there are examples in blocks test suite: 
```
it(" should error if block does not exist", async () => {
            const explorerErr = await axios.get(`${ params.explorerUrl }/block/${ params.invalidBlockHash }`).catch(response => { return response; });
            expect(explorerErr.response.status).to.equal(404);
            expect(explorerErr.response.statusText).to.equal("Not Found");
            expect(explorerErr.response.data).to.equal("block not found");
        });
```

**Review by otoumas — APPROVED** [review]
LGTM See comments, there some possible refactor proposals, not blocking!

File: src/utils/Utils.ts (hunk 1 above)
Comment (carlosper88) [reply src/utils/Utils.ts:16]: maybe in the future we'll need some non date related util methods 😅

File: test/integration/initial_sync/report.tests.ts (hunk 2 above)
Comment (carlosper88) [reply test/integration/initial_sync/report.tests.ts:39]: Yes, you're right. It's the same and shouldn't be duplicated. Gonna change it

File: src/utils/Utils.ts (hunk 1 above)
Comment (otoumas) [reply src/utils/Utils.ts:16]: Another reason to rename it, we can create another file for those other methods and keep things small and tidy 😉

**Review by otoumas — APPROVED** [review]
LGTM

**Pull request merged**